src/NLog/LayoutRenderers/ExceptionLayoutRenderer.cs

```diff
@@ -285,11 +296,37 @@ private void AppendException(Exception currentException, IEnumerable<ExceptionRe
                 int beforeRenderLength = builder.Length;
                 var currentRenderFunction = _renderingfunctions[renderingFormat];
                 currentRenderFunction(builder, currentException);
+
                 if (builder.Length == beforeRenderLength && builder.Length != orgLength)
                 {
                     builder.Length = orgLength;
                 }
             }
+
+            var exceptionType = currentException.GetType();
+            while (exceptionType != typeof(Exception))
+            {
+                if (_customRenderingfunctions.TryGetValue(exceptionType, out var renderer))
+                {
+                    try
+                    {
+                        builder.Append(Separator);
+                        var render = renderer(currentException);
+                        builder.Append(render);
+                    }
+                    catch (Exception exception)
+                    {
+                        var message = $"Exception in {typeof(ExceptionLayoutRenderer).FullName}.AppendException(): {exception.GetType().FullName}. Custom renderer for {currentException.GetType().Name} threw an exception";
+
+                        InternalLogger.Warn(exception, message);
+                    }
+                    break;
+                }
+                else
+                {
+                    exceptionType = exceptionType.BaseType;
```

Comment:
I think this doens't work for interface ;)

maybe a some test cases first?

Reply:
@304NotModified When would exceptionType ever be an interface? `currentException.GetType()` will always return an exception type representing a class so accessing BaseType hierarchically should never be an interface type.

Reply:
` public static void RegisterExceptionRenderer<IMyExceptionInterface>(...)`

Reply:
RegisterExceptionRenderer is limited to Exception subclasses at the moment:

`public static void RegisterExceptionRenderer<T>(Func<T, string> renderer) where T : Exception`

Reply:
not sure if that's a good idea. We could enumerate the list and use IsInstanceOfType? Then it also works if someone nice abstractions. :)

Reply:
If someone registers a renderer for `MyExceptionBase` and than registers for `MyException` `IsInstanceOfType` will return true for both of them

Reply:
@304NotModified Also, if the type implements two interfaces and someone registers custom renderer for both of them which one should be used ?

Reply:
Both or closest path.